server/src/main/java/org/opensearch/search/pipeline/AbstractProcessor.java

```diff
@@ -6,20 +6,20 @@
  * compatible open source license.
  */
 
-package org.opensearch.search.pipeline.common;
-
-import org.opensearch.search.pipeline.Processor;
+package org.opensearch.search.pipeline;
 
 /**
  * Base class for common processor behavior.
  */
-abstract class AbstractProcessor implements Processor {
+public abstract class AbstractProcessor implements Processor {
```

Comment:
Why this class was moved to `server`? It belongs to `search-pipeline-commons`

Reply:
it was from the [comment](https://github.com/opensearch-project/OpenSearch/pull/8373#discussion_r1249233979) that AbstractProcessor can be used to create other processors, and can be used in ingested pipeline too.

Reply:
Should it be moved to opensearch-core than? We don't want to anyone to depend on server while implementing the pipeline processors.

Reply:
@mingshl sorry, could you move it out of server?

Reply:
tagging @msfroh for consideration if we want to refactor the entire pipeline class in the future